core/src/main/java/com/linecorp/armeria/server/HttpHeaderPathMapping.java

```diff
@@ -80,11 +80,13 @@
     public PathMappingResult apply(PathMappingContext mappingCtx) {
         final PathMappingResult result = pathStringMapping.apply(mappingCtx);
         if (!result.isPresent()) {
+            if (mappingCtx.isPreflight() && !mappingCtx.delayedThrowable().isPresent()) {
+                mappingCtx.delayThrowable(HttpStatusException.of(HttpStatus.FORBIDDEN));
```

Comment:
Could you explain why we do not return `PathMappingResult.empty()` to trigger 404 here?

Reply:
Sorry for missing comment. 🙇 
I referred to https://fetch.spec.whatwg.org/#http-responses
```
In case a server does not wish to participate in the CORS protocol, its HTTP response to the CORS or CORS-preflight request must not include any of the above headers. The server is encouraged to use the 403 status in such HTTP responses.
```
So I thought `403 forbidden` seems better than `404 not found` for a preflight request.
Let me add a comment about this!